### ISSUE TITLE
[Feature Request]: Version ReadMEs just as modules are versioned

### ISSUE THREAD

**Issue by MrMCake** [Azure/ResourceModules#1442]
### Description

We implement a feature that enables us to version module ReadME files. This is important as the repository will only always contain the 'latest' ReadMe. I.e. you cannot got back and find a ReadMe for a previous module version.

One idea might be to populate a table in the Wiki that links a version with a ReadMe's link at the time of a specific commit (i.e. when a publish was executed). For example

---

### Key Vault
| Version | ReadME | Registry Reference | Template Spec Reference | Azure DevOps Artifact reference |
| - | - | - | - | - |
| `1.0.0` | [link](https://github.com/Azure/ResourceModules/blob/e42b1e9ab499ca63bfc1aca8c8f4cd953e3a759c/arm/Microsoft.KeyVault/vaults/readme.md) | `br/modules:microsoft.keyvault.vaults:1.0.0` | `ts/modules:microsoft.keyvault.vaults:1.0.0` | `https://dev.azure.com/contoso/CARML/_artifacts/feed/ResourceModules/UPack/microsoft.keyvault.vaults/overview/1.0.0` |
| `2.0.0` | [link](https://github.com/Azure/ResourceModules/blob/e42b1e9ab499ca63bfc1aca8c8f4cd953e3a759c/arm/Microsoft.KeyVault/vaults/readme.md) | `br/modules:microsoft.keyvault.vaults:2.0.0` | `ts/modules:microsoft.keyvault.vaults:2.0.0` | `https://dev.azure.com/contoso/CARML/_artifacts/feed/ResourceModules/UPack/microsoft.keyvault.vaults/overview/2.0.0` |

**Comment by MariusStorhaug** [comment]
Could create a webapp deployment that is deployed together with the bicep/ts RG. - Could likely need some assistance of a app dev to engineer this?
`- or -
Add solution for ADO to publish on Wiki, and GH to use pages.

A good reference of what this also could look like is the ARR docs, where you have a pulldown menu of "API version".

**Comment by MrMCake** [comment]
Agreed. A benefit I see with the initial suggestion is that it's not actually necessary to 'publish' a readme. The issue would instead be solved by simply maintaining the table properly.

**Comment by MrMCake** [comment]
Moved into `blocked` to not re-discuss it until the hackathon is over
